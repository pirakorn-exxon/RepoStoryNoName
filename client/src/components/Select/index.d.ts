import * as React from "react";
import { FieldProps } from "../Field/index";

export interface SelectProps extends FieldProps {
  /** Label for checkbox list*/
  label?: any;
  /** note shown*/
  note?: any;
  /** value is the value currently chosen of the select*/
  value?: any;
  /** If the Select is valid or not*/
  valid?: boolean;
  /** If the Select is error state or not*/
  error?: boolean;
  /** If the select are disabled or not*/
  disabled?: boolean;
  /** If the selec is read only or not*/
  readOnly?: boolean;
  /** Function triggered by changes to the checkbox*/
  onChange?: (...args: any[]) => any;
  /** Name for accessibility*/
  name?: string;
  /** id for accessibility*/
  id?: string;
  /** Props to send to the select element */
  selectProps?: React.SelectHTMLAttributes<HTMLSelectElement>;
  /** Additional css classes to append to the outer element*/
  className?: string;
  /** React component children, in this case Radios*/
  children?: React.ReactNode;
}

export const Select: React.FC<SelectProps>;
