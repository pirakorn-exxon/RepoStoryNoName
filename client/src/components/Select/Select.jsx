import React from "react";
import PropTypes, { oneOfType } from "prop-types";
import clsx from "clsx";
import { Field } from '@exxonmobil/react-unity';

/** Allows a user to select a value from a series of options */
const Select = props => {
  const {
    label,
    id,
    note,
    value,
    valid,
    error,
    disabled,
    onChange,
    readOnly,
    name,
    className,
    children,
    selectProps,
    selectWidth,
    ...other
  } = props;

  const { className: selectClassName = "", ...otherSelectProps } = selectProps;

  return (
    <Field
      className={clsx("em-c-field--date-picker", className)}
      error={error}
      valid={valid}
      disabled={disabled}
      readOnly={readOnly}
      {...other}
    >
      {label && <Field.Label>{label}</Field.Label>}
      <Field.Body>
        <select
          className={clsx("em-c-select", selectClassName)}
          id={id}
          onChange={onChange}
          value={value}
          disabled={disabled}
          name={name}
          style={{width: selectWidth}}
          {...otherSelectProps}
        >
          {children}
        </select>
      </Field.Body>
      {note && <Field.Note>{note}</Field.Note>}
    </Field>
  );
};

Select.propTypes = {
  /** Label for checkbox list */
  label: oneOfType([PropTypes.string, PropTypes.number]),
  /** note shown */
  note: oneOfType([PropTypes.string, PropTypes.number]),
  /** value is the value currently chosen of the select */
  value: oneOfType([PropTypes.string, PropTypes.number]),
  /** If the Select is valid or not */
  valid: PropTypes.bool,
  /** If the Select is error state or not */
  error: PropTypes.bool,
  /** If the select are disabled or not */
  disabled: PropTypes.bool,
  /** If the selec is read only or not */
  readOnly: PropTypes.bool,
  /** Function triggered by changes to the checkbox */
  onChange: PropTypes.func,
  /** Name for accessibility */
  name: PropTypes.string,
  /** id for accessibility */
  id: PropTypes.string,
  /** Props to send to the select element */
  selectProps: PropTypes.object,
  /** Additional css classes to append to the outer element */
  className: PropTypes.string,
  /** React component children, in this case Radios */
  children: PropTypes.node,
  /** Custom Select Width */
  selectWidth: PropTypes.string
};

Select.defaultProps = {
  label: "",
  note: "",
  value: undefined,
  valid: false,
  error: false,
  disabled: false,
  readOnly: false,
  onChange: () => {},
  name: "",
  id: "",
  className: "",
  selectProps: {},
  children: null,
  selectWidth: ""
};

export default Select;
