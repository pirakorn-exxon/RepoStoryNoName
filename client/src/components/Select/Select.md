```jsx
const [value, setValue] = React.useState("");
const [errorValue, setErrorValue] = React.useState("");
const [validValue, setValidValue] = React.useState("");

<Fieldset>
  <Grid variant="2-up">
    <Grid.Item>
      <Select
        label="Default"
        value={value}
        onChange={(e) => setValue(e.target.value)}
      >
        <optgroup label="Option Group">
          <option value="option-1">Option 1</option>
          <option value="option-2">Option 2</option>
        </optgroup>
      </Select>
    </Grid.Item>
    <Grid.Item>
      <Select label="Disabled" disabled>
        <optgroup label="Option Group">
          <option value="option-1">Option 1</option>
          <option value="option-2">Option 2</option>
        </optgroup>
      </Select>
    </Grid.Item>
    <Grid.Item>
      <Select
        label="Error"
        value={errorValue}
        error
        note="Invalid State"
        onChange={(e) => setErrorValue(e.target.value)}
      >
        <optgroup label="Option Group">
          <option value="option-1">Option 1</option>
          <option value="option-2">Option 2</option>
        </optgroup>
      </Select>
    </Grid.Item>
    <Grid.Item>
      <Select
        label="Valid"
        value={validValue}
        valid
        note="Perfect selection"
        onChange={(e) => setValidValue(e.target.value)}
      >
        <optgroup label="Option Group">
          <option value="option-1">Option 1</option>
          <option value="option-2">Option 2</option>
        </optgroup>
      </Select>
    </Grid.Item>
  </Grid>
</Fieldset>
```
