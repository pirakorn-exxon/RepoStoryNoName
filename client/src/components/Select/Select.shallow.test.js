import React from "react";
import { shallow } from "enzyme";
import Select from "./Select";
it("Should render Select component correctly", () => {
  const wrapper = shallow(<Select></Select>);
  expect(wrapper).toMatchSnapshot();
});