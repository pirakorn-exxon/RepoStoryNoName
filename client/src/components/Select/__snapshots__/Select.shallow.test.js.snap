// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Select component correctly 1`] = `
<Field
  className="em-c-field--date-picker "
  disabled={false}
  error={false}
  readOnly={false}
  valid={false}
>
  <FieldBody
    className=""
  >
    <select
      className="em-c-select "
      disabled={false}
      id=""
      onChange={[Function]}
    />
  </FieldBody>
</Field>
`;
